tools/celtest/test_runner.go

```diff
@@ -627,7 +652,30 @@ func (tr *TestRunner) createResultMatcherFromPB(t *testing.T, testCase *conforma
 			return failureResult
 		}, nil
 	case *conformancepb.TestOutput_Unknown:
-		//  TODO: to implement
+		// Validate that all expected unknown expression ids are returned by the evaluation result.
+		return func(out ref.Val, err error) TestResult {
+			expectedUnknownIDs := testOutput.Unknown.GetExprs()
+			if err == nil && types.IsUnknown(out) {
+				actualUnknownIDs := out.Value().(*types.Unknown).IDs()
+				if len(expectedUnknownIDs) != len(actualUnknownIDs) {
```

Comment:
could you use cmp here instead? https://pkg.go.dev/github.com/google/go-cmp/cmp#example-Diff-Testing